pyomo/contrib/matching/maximum_matching.py

```diff
@@ -0,0 +1,42 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and 
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain 
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from pyomo.common.dependencies import networkx as nx
+nxb = nx.algorithms.bipartite
+nxc = nx.algorithms.components
+from_biadjacency_matrix = nxb.matrix.from_biadjacency_matrix
```

Comment:
These cause `networkx` to be unconditionally imported when this module is imported.  You might consider moving them into the `maximal_matching()` function.